pkg/reconciler/mtbroker/trigger/trigger.go

```diff
@@ -96,6 +152,22 @@ func (r *Reconciler) reconcileTrigger(ctx context.Context, b *eventingv1.Broker,
 	return nil
 }
 
+func (r *Reconciler) trackBroker(ctx context.Context, t *eventingv1.Trigger) error {
+	trackKResource := r.kresourceTracker.TrackInNamespace(t)
+	brokerObjRef := corev1.ObjectReference{
+		Kind:       brokerGVK.Kind,
+		APIVersion: brokerGVK.GroupVersion().String(),
+		Name:       t.Spec.Broker,
+		Namespace:  t.Namespace,
+	}
+
+	if err := trackKResource(brokerObjRef); err != nil {
```

Comment:
Can this track non-existent resources?

Reply:
yes it does. We do have an e2e test I thought for creating trigger first, then creating broker. But also validated manually:

```
vaikas-a01:eventing vaikas$ kubectl -n vaikas-test get triggers
NAME                   READY   REASON               BROKER         SUBSCRIBER_URI                                       AGE
triggerwithnobroker2   False   BrokerDoesNotExist   broker99                                                            44s
```

```
vaikas-a01:eventing vaikas$ kubectl apply -f ~/demo/broker2.yaml
broker.eventing.knative.dev/broker99 created
```

```
vaikas-a01:eventing vaikas$ kubectl -n vaikas-test get triggers
NAME                   READY   REASON               BROKER         SUBSCRIBER_URI                                       AGE
triggerwithnobroker2   True                         broker99       http://event-display.vaikas-test.svc.cluster.local   63s
```